src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,13 +3,48 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int LENGTH_OF_STORAGE = 10;
+    private StorageCell<K, V>[] cell;
+    private int firstEmptyCell;
+
+    public StorageImpl() {
+        cell = new StorageCell[10];
+        firstEmptyCell = 0;
+    }
 
     @Override
     public void put(K key, V value) {
+        if (!changeValueIfKeyExist(key, value)) {
+            cell[firstEmptyCell] = new StorageCell<>();
+            cell[firstEmptyCell].setKey(key);
+            cell[firstEmptyCell].setValue(value);
+            firstEmptyCell++;
+        }
     }
 
     @Override
     public V get(K key) {
+        for (StorageCell<K, V> tempCell : cell) {
+            if (tempCell == null) {
+                return null;
+            }
+            if (tempCell.getKey() == null ? key == null : tempCell.getKey().equals(key)) {
```

Comment:
```suggestion
            if (tempCell.getKey() ==  key || key != null && key.equals(tempCell.getKey())) {
```
make same changes to other checks